centaurCwlRunner/src/main/scala/centaur/cwl/Outputs.scala

```diff
@@ -0,0 +1,106 @@
+          case Right(Right(cwl)) =>
+
+            CentaurCromwellClient.outputs(submittedWorkflow).get.outputs match {
+              case JsObject(map) =>
+                val typeMap: Map[String, MyriadOutputType] = cwl.fold(CwlOutputsFold)
+                val mungeTypeMap = typeMap.mapKeys(stripTypeMapKey)
+
+                val mungeOutputMap = map.mapKeys(stripOutputKey)
+
+                mungeOutputMap.
+                  //This lets us operate on the values of the output values and types for a particular output key
+                  intersectWith(mungeTypeMap)(resolveOutput).
+                  //converting the whole response to Json using Circe's auto-encoder derivation
+                  asJson.
+                  //drop null values so that we don't print when Option == None
+                  pretty(io.circe.Printer.spaces2.copy(dropNullValues = true))
+              case other => s"it seems cromwell is not returning outputs as a Jsobject but is instead a $other"
+            }
+          case Right(Left(error)) => s"couldn't parse $some: $error"
+          case Left(error) => s"couldn't parse $some: $error"
+        }
+      case Some(other) => s"received the value $other when the workflow string was expected"
+      case None => "the workflow is no longer in the metadata payload, it's a problem"
+    }
+  }
+
+  //Ids come out of SALAD pre-processing with a filename prepended.  This gets rid of it
+  def stripTypeMapKey(key: String): String = key.substring(key.lastIndexOf("#") + 1, key.length)
+
+  //Ids come out of Cromwell with a prefix, separated by a ".".  This takes everything to the right,
+  //as CWL wants it
+  def stripOutputKey(key: String): String = key.substring(key.lastIndexOf(".") + 1, key.length)
+
+  //In an Ideal world I'd return a Coproduct of these types and leave the asJson-ing to the handleOutput
+  def resolveOutput(jsValue: JsValue, mot: MyriadOutputType): Json  = {
+    (jsValue, mot) match {
+      //CWL expects quite a few enhancements to the File structure, hence...
+      case (JsString(metadata), Inl(CwlType.File)) =>
+
+        val fileName = better.files.File(metadata)
+
+        val file = fs2.io.file.readAll[IO](fileName.path, 65536)
+        val hash = fs2.hash.sha1[IO]
+
+        val bytes = (file through hash).runLog.unsafeRunSync
+
+        //hexify the bytes into a string
+        val hex = bytes.map {
+          b => String.format("%02X", new Integer(b & 0xff))
+        }.mkString.toLowerCase
+
+        CwlFile(
+          location = Option(fileName.name),
+          checksum = Option("sha1$" + hex),
+          size = Option(fileName.size)
+        ).asJson
+      case (JsNumber(metadata), Inl(CwlType.Long)) => metadata.longValue.asJson
+      case (JsNumber(metadata), Inl(CwlType.Float)) => metadata.floatValue().asJson
+      case (JsNumber(metadata), Inl(CwlType.Double)) => metadata.doubleValue().asJson
+      case (JsNumber(metadata), Inl(CwlType.Int)) => metadata.intValue().asJson
```

Comment:
why do these have parens

Reply:
because they're 2-ples?

Reply:
On the right side, ex: `metadata.intValue().asJson` should be `metadata.intValue.asJson`

Reply:
tuples

Reply:
ah I will clean up the right side, thanks @kshakir